Makefile

```diff
@@ -20,7 +20,7 @@ SHELL:=/usr/bin/env bash
 
 .DEFAULT_GOAL:=help
 
-GO_VERSION ?= 1.20.4
+GO_VERSION ?= 1.20.8
```

Comment:
Same as other PR. This is not bumping Go.

/hold

Reply:
followed this PR: https://github.com/kubernetes-sigs/cluster-api/pull/9381 
I will check again if we need any other changes.

Reply:
Our build is different than CAPI. We don't utilize this variable for anything. In CAPI it is propagated everywhere.